File: php/features.php
<?php
declare(strict_types=1);

// Mock SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class MockFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new MockBaseFeature();
            case "test":
                return new MockTestFeature();
            default:
                return new MockBaseFeature();
        }
    }
}

File: py/features.py
# Mock SDK feature factory

from feature.base_feature import MockBaseFeature
from feature.test_feature import MockTestFeature


def _make_feature(name):
    features = {
        "base": lambda: MockBaseFeature(),
        "test": lambda: MockTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
